nvflare/private/fed/app/deployer/simulator_deployer.py

```diff
@@ -76,11 +76,12 @@ def create_fl_client(self, client_name, args):
         federated_client = deployer.create_fed_client(args)
 
         self._create_client_cell(client_config, client_name, federated_client)
+        client_engine = SimulatorParentClientEngine(federated_client, args)
+        federated_client.set_client_engine(client_engine)
 
-        federated_client.register()
+        with client_engine.new_context() as fl_ctx:
+            federated_client.register({}, fl_ctx)
```

Comment:
minor stuff, it might be more clear to spell out what is the argument name,

`federated_client.register(register_data={}, fl_ctx=fl_ctx)`

And we need to update the docstring in the register method so people know what this "register_data" is about.

Reply:
done